rust/matches/src/lib.rs

```diff
@@ -623,6 +626,146 @@ pub mod matches {
 
         Ok(())
     }
+
+    pub fn match_join_namespace<'a, 'b, 'c, 'info>(
+        ctx: Context<'a, 'b, 'c, 'info, MatchJoinNamespace<'info>>,
+    ) -> Result<()> {
+        if !is_namespace_program_caller(&ctx.accounts.instructions.to_account_info()) {
+            return Err(error!(ErrorCode::UnauthorizedCaller));
+        }
+
+        let match_instance = &mut ctx.accounts.match_instance;
+
+        let namespaces = match match_instance.namespaces.clone() {
+            Some(namespaces) => namespaces,
+            None => return Err(error!(ErrorCode::FailedToJoinNamespace)),
+        };
+
+        let mut joined = false;
+        let mut new_namespaces = vec![];
+        for mut ns in namespaces {
+            if ns.namespace == anchor_lang::solana_program::system_program::id() && !joined {
+                ns.namespace = ctx.accounts.namespace.key();
+                ns.index = None;
+                ns.inherited = InheritanceState::NotInherited;
+                joined = true;
+                new_namespaces.push(ns);
+            } else {
```

Comment:
kill the else and move the push to outside the remaining if

Reply:
It's either this or use a continue, imo i think the else is more clear

Reply:
I see that you now updated this to not mutate and to do a new variable in the if instead. That's much better then what it was when I commented. My comment was when it was just a mutation and `continue` would not have been necessary.